pkg/dockerregistry/server/repositorymiddleware.go

```diff
@@ -137,20 +150,25 @@ func (r *repository) Blobs(ctx context.Context) distribution.BlobStore {
 	repo := repository(*r)
 	repo.ctx = ctx
 
-	bs := &quotaRestrictedBlobStore{
-		BlobStore: r.Repository.Blobs(ctx),
-		repo:      &repo,
-	}
-	if !r.pullthrough {
-		return bs
-	}
+	bs := r.Repository.Blobs(ctx)
 
-	return &pullthroughBlobStore{
+	bs = &quotaRestrictedBlobStore{
 		BlobStore: bs,
 
-		repo:          &repo,
-		digestToStore: make(map[string]distribution.BlobStore),
+		repo: &repo,
 	}
+
+	if r.pullthrough {
```

Comment:
We could merge the wrappers into one for sure. But actually I find it more synoptical this way. The functionality is 
grouped by class and each class can be easily turned off by configuration option. The resulting blob store can easily 
end up with no blobstore wrappers
On 05/26/2016 10:51 AM, Maciej Szulik wrote:
>
> In pkg/dockerregistry/server/repositorymiddleware.go <https://github.com/openshift/origin/pull/8195#discussion_r64711285>:
>
> >  	}
> > +
> > +	if r.pullthrough {
>
> I'll leave it here to hang ;) Do we care about quota when doing pullthrough? I don't think so, from what you've told 
> me we don't store anything doing so, why adding that unnecessary layer here? Or am I missing something here?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub 
> <https://github.com/openshift/origin/pull/8195/files/779e0326863b0e5a648bc20a66c594b947edf73c..0094ab66c3d7bf7a4550595baf4286cc561f9f27#r64711285>
>